### PR TITLE
Complete homework 1

### PR DESCRIPTION
**Pull request by c1ydehhx**
> A good PR should include the following parts:
> - [x] Provide a clear title
> - [x] Provide meaningful descriptions in the PR and the commit logs
> - [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
> - [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
> - [x] Exclude irrelevant files, such as binaries, text files, or dot files
> - [x] Make sure tests/CI pass

Hi. In this PR, I already completed the homework 1 and get passed by GitHub Actions.

## Implement

 - [x] Create a new folder with username (e.g. `ntut-xuan` for me).
 - [x] Copy `countline.py` to the new folder.
 - [x] Create `countline.sh` for writing shell script.
 - [x] Implement `countline.sh` shell script and have equality feature of `countline.py`
 - [x] Adjust python script to support `PYTHON_BIN` environment variable. The script can decide by environment variable to determine which python binary should be used. (See `countline.py` diff for more information)
 - [x] Check the `validate.sh` script can run `countline.sh` property.

## How to review it

 - [x] Check the GitHub Actions passed and show **5 of `GET POINT 1`**.

### CODE REVIEW

**Review by ThreeMonth03 — COMMENTED** [review]
https://github.com/yungyuc/nsdhw_24au/blob/21090c9f719bdfafa503131fa6f3b786de949459/hw1/README.rst?plain=1#L5-L5 
If you finish your homework, you need to submit the url of this pr on [new E3](https://e3p.nycu.edu.tw/my/). 
image

I will review this pr after you submit this url to E3.

**Comment by c1ydehhx** [comment]
> https://github.com/yungyuc/nsdhw_24au/blob/21090c9f719bdfafa503131fa6f3b786de949459/hw1/README.rst?plain=1#L5-L5
> 
> 
> If you finish your homework, you need to submit the url of this pr on [new E3](https://e3p.nycu.edu.tw/my/).
> image
> I will review this pr after you submit this url to E3.

I see. Thanks for the notice.

**Review by ThreeMonth03 — APPROVED** [review]
LGTM.

File: hw0/README.rst
Comment (yungyuc) [inline hw0/README.rst:1]: It is wrong to delete hw0.  @ntut-xuan @ThreeMonth03 please help to fix it in the `hw1` branch.

**Review by yungyuc — COMMENTED** [review]
It is wrong to delete hw0.  @ntut-xuan @ThreeMonth03 please help to fix it in the `hw1` branch.

**Comment by yungyuc** [comment]
It's better to have the changesets that wrongly deleted files squashed.  Please do so in the future.

**Comment by ThreeMonth03** [comment]
@ntut-xuan By the way, your grade is recorded to E3, if you have any question, please tell me as soon as possible.

**Pull request merged**